lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3029,14 +3029,47 @@ bool ConformanceChecker::checkActorIsolation(
   bool isDistributed = refResult.isolation.isDistributedActor() &&
       !witness->getAttrs().hasAttribute<NonisolatedAttr>();
   if (isDistributed) {
-    // If we're coming from a non-distributed requirement, then the requirement
-    // must be 'throws' to accommodate failure.
-    if (!isDistributedDecl(requirement) && !isThrowsDecl(requirement))
-      missingOptions |= MissingFlags::RequirementThrows;
-
-    if (!isDistributedDecl(witness) &&
-        (isDistributedDecl(requirement) || !missingOptions))
-      missingOptions |= MissingFlags::WitnessDistributed;
+    // Check if the protocol where the requirement originates from
+    // is a distributed actor constrained one.
+    auto proto = dyn_cast<ProtocolDecl>(requirement->getDeclContext());
+    if (proto && proto->isDistributedActor()) {
```

Comment:
Requirements are always within protocols, FWIW, so you could use `cast<ProtocolDecl>` and not check for null.

Reply:
Sounds right thanks!